rslib/src/import_export/text/import.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::{collections::HashMap, sync::Arc};
+
+use crate::{
+    import_export::{
+        text::{ForeignCard, ForeignData, ForeignNote, ForeignNotetype, ForeignTemplate},
+        LogNote, NoteLog,
+    },
+    notetype::{CardGenContext, CardTemplate, NoteField, NotetypeConfig},
+    prelude::*,
+};
+
+impl ForeignData {
+    pub fn import(self, col: &mut Collection) -> Result<OpOutput<NoteLog>> {
+        col.transact(Op::Import, |col| {
+            let mut ctx = Context::new(&self, col)?;
+            ctx.import_foreign_notetypes(self.notetypes)?;
+            ctx.import_foreign_notes(self.notes)
+        })
+    }
+}
+
+struct Context<'a> {
+    col: &'a mut Collection,
+    notetypes: HashMap<String, Option<Arc<Notetype>>>,
+    deck_ids: HashMap<String, Option<DeckId>>,
+    usn: Usn,
+    normalize_notes: bool,
+    //progress: IncrementableProgress<ImportProgress>,
+}
+
+impl<'a> Context<'a> {
+    fn new(data: &ForeignData, col: &'a mut Collection) -> Result<Self> {
+        let usn = col.usn()?;
+        let normalize_notes = col.get_config_bool(BoolKey::NormalizeNoteText);
+        let mut notetypes = HashMap::new();
+        notetypes.insert(
+            String::new(),
+            col.notetype_for_string(&data.default_notetype)?,
+        );
```

Comment:
This zero-value-as-default approach is a bit subtle, and might be worth documenting briefly in the struct's field docstring.